lib/_http_outgoing.js

```diff
@@ -299,6 +300,11 @@ function storeHeader(self, state, field, value) {
   // minimize the performance impact in the common case.
   if (/[\r\n]/.test(value))
     value = value.replace(/[\r\n]+[ \t]*/g, '');
+  // Protect against invalid header field names... but only in strictmode
+  if (self.strictMode && !common._checkIsHttpToken(field, 'Header name')) {
```

Comment:
Ah, good catch. That was an artifact from a previous edit. I originally had the method raising the error itself but thought better of it.